spec/sidekiq_cli_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'open3'
+
+describe 'sidekiq CLI' do
+  it 'runs without errors' do
+    # Inspired by: https://github.com/sidekiq/sidekiq/issues/3214
+    # Kick off sidekiq, wait a bit, and make sure the output doesn't include errors.
+    # It's slow, but appears to be the only way to detect errors outside the workers.
+    errors = Open3.popen2e('bundle exec sidekiq') do |_stdin, stdout_and_stderr, wait_thread|
+      sleep 30
```

Comment:
I'm fine with keeping this at 30 seconds, but I wonder if we could reduce it so that it doesn't slow down CI too much.

Reply:
I considered reducing it, but I want to be sure to catch any issues that might take a little while to pop up. Also, the whole spec suite still only takes just over a minute to run, so it's not too bad.

...that said, I think that it will bug me if this isn't optimized just a bit. I'm going to tweak it to speed things up both locally and in CI. Stay tuned...